opensquirrel/merger/general_merger.py

```diff
@@ -75,18 +80,19 @@ def merge_single_qubit_gates(squirrel_ir: SquirrelIR):
             composed = compose_bloch_sphere_rotations(statement, already_accumulated)
             accumulators_per_qubit[statement.qubit] = composed
 
-            del squirrel_ir.statements[statement_index]
+            del ir.statements[statement_index]
             continue
 
         for qubit_operand in statement.get_qubit_operands():
             if not accumulators_per_qubit[qubit_operand].is_identity():
-                squirrel_ir.statements.insert(statement_index, accumulators_per_qubit[qubit_operand])
+                ir.statements.insert(statement_index, accumulators_per_qubit[qubit_operand])
                 accumulators_per_qubit[qubit_operand] = BlochSphereRotation.identity(qubit_operand)
                 statement_index += 1
         statement_index += 1
 
     for accumulated_bloch_sphere_rotation in accumulators_per_qubit.values():
         if not accumulated_bloch_sphere_rotation.is_identity():
-            squirrel_ir.statements.append(accumulated_bloch_sphere_rotation)
-
-    squirrel_ir.statements = sorted(squirrel_ir.statements, key=lambda obj: isinstance(obj, Measure))
+            default_bloch = accumulated_bloch_sphere_rotation.get_default_bloch()
+            if default_bloch is not None:
```

Comment:
If a `BlochSphereRotation` is returned (no None), you can delete this check.

Reply:
Beautiful, yes!